### ISSUE TITLE
Bug when using accessibility features

### ISSUE THREAD

**Issue by RadkevichDenis** [meliorence/react-native-render-html#558]
### Decision Table

- [X] My issue does not look like “The HTML attribute 'xxx' is ignored” (unless we claim support for it)
- [X] My issue does not look like “The HTML element `<yyy>` is not rendered”

### Good Faith Declaration

- [X] I have read the HELP document here: https://git.io/JBi6R
- [X] I have read the CONTRIBUTING document here: https://git.io/JJ0Pg
- [X] I have confirmed that this bug has not been reported yet

### Description

When adding a link inside a `div` block and with Voice Over(iOS) or TalkBack(Android) enabled, the link cannot be clicked. Doesn't work only for certain text lengths or certain font sizes.

If the text before the link is short (2-3 words), then there is no problem, with the same size of the text of the `link` and the `div` text. But when increasing the length of the text before the link, the `onPress` function stops working in a11y mode.

### React Native Information

```sh
OS: macOS 11.2.2
    CPU: (6) x64 Intel(R) Core(TM) i5-8500B CPU @ 3.00GHz
    Memory: 213.37 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.13.0 - ~/.nvm/versions/node/v16.13.0/bin/node
    Yarn: 1.22.11 - /usr/local/bin/yarn
    npm: 8.1.0 - ~/.nvm/versions/node/v16.13.0/bin/npm
    Watchman: 4.9.0 - /opt/local/bin/watchman
  Managers:
    CocoaPods: 1.10.1 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.5, DriverKit 20.4, macOS 11.3, tvOS 14.5, watchOS 7.4
    Android SDK: Not Found
  IDEs:
    Android Studio: 4.1 AI-201.8743.12.41.6953283
    Xcode: 12.5/12E262 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.9 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.68.0 => 0.68.0 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

### RNRH Version

6.3.4

### Tested Platforms

- [X] Android
- [X] iOS
- [ ] Web
- [ ] MacOS
- [ ] Windows

### Reproduction Platforms

- [X] Android
- [X] iOS
- [ ] Web
- [ ] MacOS
- [ ] Windows

### Minimal, Reproducible Example

https://github.com/RadkevichDenis/RNRH-a11y-issue

### Additional Notes
The issue
https://user-images.githubusercontent.com/44681387/161560079-83f9f5b2-46bb-4435-be3c-7cfe5ab23bcd.mp4

Font-Size: 2(works as expected)
https://user-images.githubusercontent.com/44681387/161560125-8a0cdc28-a932-4f70-9b6a-b07d4c99f176.mp4

Small text(works as expected)
https://user-images.githubusercontent.com/44681387/161560184-450ca60f-5cc2-4f97-85b0-0f4f34f80a3c.mp4

**Comment by jsamr** [comment]
This is a React Native bug, nothing actionable for us here! I know there have been a lot of A11y fixes recently in RN, it's possible upgrading to latest will fix this peculiar issue.
